tslib/react/src/components/DataGrid.tsx

```diff
@@ -12,372 +18,364 @@ import {
-  rows: T[]
-  keyField: keyof T
-  dense?: boolean
-  collapseBody?: (rowIndex: number) => React.ReactNode
-  initialRowsPerPage?: number
-  rowsPerPageOption?: Array<number | { value: number; label: string }>
-  defaultFilter?: (row: T) => boolean
+function DataGrid<T>({
+  data,
+  columns,
+}: {
+  data: T[]
+  columns: ColumnDef<T>[]
 }): React.ReactElement {
-  const { columns, rows, keyField, dense, collapseBody, defaultFilter } = props
-  let { initialRowsPerPage, rowsPerPageOption } = props
-  const [order, setOrder] = React.useState<Order>("asc")
-  const [orderBy, setOrderBy] = React.useState<number>(0) // index of columns
-  const [page, setPage] = React.useState(0)
-  const [filters, setFilters] = React.useState<RowFilter[]>([])
-
-  const getRowIndex = (row: T): number => {
-    return rows.findIndex((row2) => row[keyField] === row2[keyField])
-  }
-
-  // Pagination
-  rowsPerPageOption = rowsPerPageOption || defaultRowsPerPageOption
-  initialRowsPerPage = initialRowsPerPage // use first element as default
-    ? initialRowsPerPage
-    : isNumber(rowsPerPageOption[0])
-      ? rowsPerPageOption[0]
-      : rowsPerPageOption[0].value
-  const [rowsPerPage, setRowsPerPage] = React.useState(initialRowsPerPage)
-
-  const handleChangePage = (_event: unknown, newPage: number) => {
-    setPage(newPage)
-  }
-
-  const handleChangeRowsPerPage = (
-    event: React.ChangeEvent<HTMLInputElement>
-  ) => {
-    setRowsPerPage(parseInt(event.target.value, 10))
-    setPage(0)
-  }
-
-  // Filtering
-  const fieldAlreadyFiltered = (columnIdx: number): boolean =>
-    filters.some((f) => f.columnIdx === columnIdx)
-
-  const handleClickFilterCell = (columnIdx: number, value: Value) => {
-    if (fieldAlreadyFiltered(columnIdx)) {
-      return
-    }
-    const newFilters = [...filters, { columnIdx: columnIdx, value: value }]
-    setFilters(newFilters)
-  }
-
-  const clearFilter = (columnIdx: number): void => {
-    setFilters(filters.filter((f) => f.columnIdx !== columnIdx))
-  }
-
-  const filteredRows = rows.filter((row, rowIdx) => {
-    if (defaultFilter?.(row)) {
-      return false
-    }
-    return filters.length === 0
-      ? true
-      : filters.some((f) => {
-          if (columns.length <= f.columnIdx) {
-            console.log(
-              `columnIdx=${f.columnIdx} must be smaller than columns.length=${columns.length}`
-            )
-            return true
-          }
-          const toCellValue = columns[f.columnIdx].toCellValue
-          if (toCellValue !== undefined) {
-            return toCellValue(rowIdx) === f.value
-          }
-          const field = columns[f.columnIdx].field
-          return row[field] === f.value
-        })
+  const [sorting, setSorting] = React.useState<SortingState>([])
+  const [columnFilters, setColumnFilters] = React.useState<ColumnFiltersState>(
+    []
+  )
+  const [pagination, setPagination] = React.useState<PaginationState>({
+    pageIndex: 0,
+    pageSize: 50,
```

Comment:
You set `pageSize` here 50 as constant, but how about receiving `initialRowsPerPage` as a prop and using that value here, as current `DataGrid` does?